src/components/FormPills/FormPills.vue

```diff
@@ -77,13 +97,19 @@ const splitItem = (item: string) => item.split(props.separator);
 const removeItem = (index: number) => items.value.splice(index, 1);
 
 const addItem = (item: string) => {
-    const newItems = splitItem(item).filter(v => v !== '');
+    if (items.value.length >= props.maxItems) {
+        return false;
+    }
```

Comment:
Feels a little weird to still allow users to type when max has been reached. I think we need to prevent user from typing and parent will already know if max has been reached.